projects/reservoir/fv3/era_process.py

```diff
@@ -108,26 +149,21 @@ def check_time_steps_complete(path, var):
     logger.info("all time steps checked.")
 
 
-def merge_into_file(path, var, day_of_week_index, day):
-    # create one file per weekday
-    for f in glob.glob(path + var + "/downloaded/*.nc"):
-        dataset = xr.open_dataset(f)
-        data_at_day = dataset.isel(time=slice(day_of_week_index, None)).isel(
-            time=slice(None, None, 7)
-        )
-        dataset.close()
-
-    sorted_data = data_at_day.sortby("time")
-    sorted_data.to_netcdf(path + var + "/merged/" + var + "_" + day + ".nc")
+def merge_into_weekly_file(path, var, args):
+    merged_data = xr.concat(
+        [xr.open_dataset(f) for f in glob.glob(path + var + "/downloaded/*.nc")],
+        dim="time",
+    )
+    merged_data = merged_data.sortby("time")
+    # get time chunk according to inout args todo
+    sorted_data = merged_data.sel(time=slice(args.start_date, args.end_date))
+    # merge into weekly mean
+    weekly_merged = calculate_weekly_mean(sorted_data)
     current_working_directory = os.getcwd()
+    weekly_merged.to_netcdf(path + var + "/merged/" + var + ".nc")
```

Comment:
Can define this netcdf name as a variable since it's used again later in the subsequent function call. For extra clarity I'd add a `weekly_mean` descriptor to the filenames.